scheduler/plugin/plugin.go

```diff
@@ -187,6 +208,48 @@ func (pl *wasmPlugin) Name() string {
 	return PluginName
 }
 
+var _ framework.EnqueueExtensions = (*wasmPlugin)(nil)
+
+// allClusterEvents is copied from framework.go, to avoid the complexity of
+// conditionally implementing framework.EnqueueExtensions.
+var allClusterEvents = []framework.ClusterEvent{
+	{Resource: framework.Pod, ActionType: framework.All},
+	{Resource: framework.Node, ActionType: framework.All},
+	{Resource: framework.CSINode, ActionType: framework.All},
+	{Resource: framework.PersistentVolume, ActionType: framework.All},
+	{Resource: framework.PersistentVolumeClaim, ActionType: framework.All},
+	{Resource: framework.StorageClass, ActionType: framework.All},
+}
+
+// EventsToRegister implements the same method as documented on framework.EnqueueExtensions.
+func (pl *wasmPlugin) EventsToRegister() (clusterEvents []framework.ClusterEvent) {
+	// TODO: chase https://github.com/kubernetes/kubernetes/pull/119155 for api changes
+
+	// Note: EventsToRegister() doesn't pass a context
+	ctx := context.Background()
+	podUID := types.UID("") // Note: we don't know the pod UID yet
```

Comment:
I think maybe this is not the way to go. @sanposhiho I think this is before the scheduling cycle, and if correct, we need to run "anonymously" or something..

Reply:
changed